system/Controller.php

```diff
@@ -123,7 +120,11 @@ public function initController(RequestInterface $request, ResponseInterface $res
 			$this->forceHTTPS($this->forceHTTPS);
 		}
 
-		$this->loadHelpers();
+		// Autoloading helper files.
+		if (! empty($this->helpers))
+		{
+			helper($this->helpers);
+		}
```

Comment:
I agree on deprecating it first before outright remove. I may be using it in a controller code elsewhere, then suddenly my application will break for unknown method.

Furthermore, the `empty` check can be removed, since `helper` already manages loading via arrays thru `foreach`. Executing `foreach` on empty arrays will just do nothing.

```suggestion
		helper($this->helpers);
```

Reply:
I think we need to keep check if helpers property not empty to avoid running helper method script and do unnecessary loads.

Reply:
The `helper()` function has no default value so checking is required.

Reply:
I don't follow. `$this->helpers` is already initialized with an empty array. So passing this directly to `helper` function will just do nothing. Am I missing something?

Reply:
Oh... you're right, this time I missed something 😅

Reply:
> I think we need to keep check if helpers property not empty to avoid running helper method script and do unnecessary loads.

I don't think it is an unnecessary load as the common functions are already loaded long before the controller is instantiated.